softwareComponents/rofiHalSimPy/hal.cpp

```diff
@@ -0,0 +1,90 @@
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+#include <pybind11/functional.h>
+
+#include <rofi_hal.hpp>
+
+namespace py = pybind11;
+
+using namespace rofi::hal;
+
+PYBIND11_MODULE( pyRofiHal, m ) {
+    m.doc() = "RoFI simulator HAL";
+
+    py::class_< Joint > joint( m, "Joint" );
+    joint
+        .def_property_readonly( "maxPosition", &Joint::maxPosition )
+        .def_property_readonly( "minPosition", &Joint::minPosition )
+        .def_property_readonly( "maxSpeed", &Joint::maxSpeed )
+        .def_property_readonly( "minSpeed", &Joint::minSpeed )
+        .def_property_readonly( "maxTorque", &Joint::maxTorque )
+        .def_property( "velocity", &Joint::getVelocity, &Joint::setVelocity )
+        .def_property( "torque", &Joint::getTorque, &Joint::setTorque )
+        .def_property_readonly( "position", &Joint::getPosition )
+        .def( "setPosition", &Joint::setPosition )
+        .def( "onError", &Joint::onError );
+
+    py::enum_< Joint::Error >( joint, "Error" )
+        .value( "Communication", Joint::Error::Communication )
+        .value( "Hardware", Joint::Error::Hardware )
+        .export_values();
+
+    py::enum_< ConnectorPosition >( m, "ConnectorPosition" )
+        .value( "Retracted", ConnectorPosition::Retracted )
+        .value( "Extended", ConnectorPosition::Extended )
+        .export_values();
+
+    py::enum_< ConnectorOrientation >( m, "ConnectorOrientation" )
+        .value( "North", ConnectorOrientation::North )
+        .value( "East", ConnectorOrientation::East )
+        .value( "South", ConnectorOrientation::South )
+        .value( "West", ConnectorOrientation::West )
+        .export_values();
+
+    py::enum_< ConnectorLine >( m, "ConnectorLine" )
+        .value( "Internal", ConnectorLine::Internal )
+        .value( "External", ConnectorLine::External )
+        .export_values();
+
+    py::enum_< ConnectorEvent >( m, "ConnectorEvent" )
+        .value( "Connected", ConnectorEvent::Connected )
+        .value( "Disconnected", ConnectorEvent::Disconnected )
+        .value( "ConnectedPower", ConnectorEvent::ConnectedPower )
+        .value( "DisconnectedPower", ConnectorEvent::DisconnectedPower )
+        .export_values();
+
+    py::class_< ConnectorState >( m, "ConnectorState" )
+        .def_readwrite( "position", &ConnectorState::position )
+        .def_readwrite( "internal", &ConnectorState::internal )
+        .def_readwrite( "external", &ConnectorState::external )
+        .def_readwrite( "connected", &ConnectorState::connected )
+        .def_readwrite( "orientation", &ConnectorState::orientation );
+
+    py::class_< Connector >( m, "Connector" )
+        .def_property_readonly( "getState", &Connector::getState )
```

Comment:
Maybe change the name of property to `state` for consistency

Reply:
Of course, the property is much better.